src/containers/app.jsx

```diff
@@ -51,6 +57,43 @@ class App extends Component {
         if (prevProps.userId !== this.props.userId) {
             storage.userId = this.props.userId;
         }
+
+        if (prevProps.installOnLoad !== this.props.installOnLoad && this.props.installOnLoad) {
+            this.offlineSupport();
+        }
+    }
+
+    async offlineSupport() {
+        this.props.startInstall();
+        // register service worker
+        if ('serviceWorker' in navigator) {
+            const wb = new Workbox('/service-worker.js');
+
+            // not 100% sure about timing here. maybe we could register the event only if it has not yet
+            // been installed. But not 100% sure..better play safe here :S
+            const installedPromise = new Promise((resolve, reject) => {
+                wb.addEventListener('installed', (event) => {
+                    resolve();
+                });
+            });
+
+            try {
+                const result = await wb.register();
+                let waitInstalled;
+                if (result.installing === null) {
+                    waitInstalled = Promise.resolve();
+                } else {
+                    waitInstalled = installedPromise;
+                }
+                await waitInstalled;
+                this.props.setInstalled();
+            } catch (e) {
+                this.props.failInstall(e);
+                throw e;
```

Comment:
This will result in uncaught exceptions higher up the tree. I'd rather not throw here, since you already reported the error to the user